dotCMS/src/integration-test/java/com/dotcms/storage/StoragePersistenceAPITest.java

```diff
@@ -0,0 +1,347 @@
+        final File imageFile = new File(
+                ConfigTestHelper.getUrlToTestResource(TEST_IMAGE_JPG).toURI());
+
+        return new Object[]{
+
+                new TestCase(StorageType.FILE_SYSTEM,
+                        groupName, textFilePath, tempFile
+                ),
+                new TestCase(StorageType.FILE_SYSTEM,
+                        groupName, imageFilePath, imageFile
+                ),
+
+                new TestCase(StorageType.DB,
+                        groupName, textFilePath, tempFile
+                ),
+                new TestCase(StorageType.DB,
+                        groupName, imageFilePath, imageFile
+                )
+
+        };
+    }
+
+    /**
+     * Test Scenario: We simply do two consecutive calls to createGroup with the same arguments
+     * Expected Results: Nothing should break. Second call should tell the group  already exist.
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("getGroupNameTestCases")
+    public void Test_Attempt_Create_Dupe_Group(final TestCase testCase) throws DotDataException {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        final String groupName = testCase.groupName;
+        assertFalse(storage.existsGroup(groupName));
+        assertTrue(storage.createGroup(groupName));
+        assertFalse(storage.createGroup(groupName));
+    }
+
+    @DataProvider
+    public static Object[] getGroupNameTestCases() {
+        final String groupName = RandomStringUtils.randomAlphanumeric(10);
+
+        return new Object[]{
+                new TestCase(StorageType.FILE_SYSTEM, groupName),
+                new TestCase(StorageType.DB, groupName)
+        };
+    }
+
+    /**
+     * Test Scenario: Regardless of group we attempt saving a binary
+     * Expected Results: We're basically testing we no longer have an exception where the binary would be inserted as a new file generating an constrain violation
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("getDupeBinaryTestCases")
+    public void Test_Attempt_Push_Same_Binary_Twice(final TestCase testCase)
+            throws DotDataException {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        storage.deleteObject(testCase.groupName, testCase.path);
+        assertTrue(storage.createGroup(testCase.groupName));
+        final Object object = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object);
+
+        assertTrue(storage.existsObject(testCase.groupName, testCase.path));
+
+        final Object object2 = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object2);
+    }
+
+    @DataProvider
+    public static Object[] getDupeBinaryTestCases() throws Exception{
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)